v6.1: update the description for tidb_enable_outer_join_reorder (#15699)
pingcap/docs: system-variables.md

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Default value: In v6.1.0, the default value is `ON`.
- Since v6.1.0, the [Join Reorder](/join-reorder.md) algorithm of TiDB supports Outer Join. This variable controls the support behavior. The default value is `OFF`, which means the Join Reorder's support for Outer Join is disabled by default.

> - If the TiDB version before the upgrade is earlier than v6.1.0, the default value of this variable after the upgrade is `ON`.

Author: The above line reuses the translation in the dev version: https://docs.pingcap.com/tidb/dev/system-variables#tidb_enable_outer_join_reorder-new-in-v610

> - If the TiDB version before the upgrade is earlier than v6.1.0, the default value of this variable after the upgrade is `ON`.

Author: ```suggestion
- If your TiDB cluster is upgraded from an earlier version to v6.1.0,  the default value of this variable after the upgrade is `ON`.
```

> - If your TiDB cluster is upgraded from an earlier version to v6.1.0, the default value of this variable after the upgrade is `ON`.

Author: ```suggestion
- If your TiDB cluster is upgraded from an earlier version to v6.1.0, the default value of this variable after the upgrade is `ON`.  If your TiDB cluster is upgraded from an earlier version to v6.1.1 or a later 6.1 patch version, the default value of this variable after the upgrade is `OFF`.
```
